### PR TITLE
fix bwd batchnorm with original layout param

### PR DESCRIPTION
**Pull request by ce1adon**

### CODE REVIEW

**Review by atamazov — COMMENTED** [review]
I do not see anything bad in this (but I do not know this domain quite well).

File: src/ocl/batchnormocl.cpp [hunk 1]
```diff
@@ -976,8 +976,8 @@ void BatchNormBackward(Handle& handle,
                 }
                 else
                 {
-                    xlocalsize = 512;
-                    xgridsize  = 512 * c;
+                    xlocalsize = 256;
+                    xgridsize  = 256 * c;
```
[the remark this answers is not in the document]
Comment (ce1adon) [reply src/ocl/batchnormocl.cpp:980]: @zpwenjh What is the error msg you saw?

[the remark this answers is not in the document]
Comment (zpwenjh) [reply src/ocl/batchnormocl.cpp:980]: @ce1adon dscale verification failed.

[the remark this answers is not in the document]
Comment (ce1adon) [reply src/ocl/batchnormocl.cpp:980]: > @ce1adon dscale verification failed.

Thanks! Our observations are consistent. PR https://github.com/zpwenjh/MIOpen/pull/14 fixed some case. I'm working on the rest.

**Comment by atamazov** [comment]
@zpwenjh If you agree with changes made by @ce1adon then could you please review and merge this PR? https://github.com/ROCmSoftwarePlatform/MIOpen/pull/916 will be automatically updated after that, and we'll be able to continue reviewing there.

**Review by atamazov — APPROVED** [review]

**Comment by ce1adon** [comment]
@atamazov Some more work are needed here.

**Comment by zpwenjh** [comment]
> @atamazov Some more work are needed here.

@ce1adon @atamazov Ok, In order to test as soon as possible, I'll merge once the bug is fixed. Thanks.

**Comment by atamazov** [comment]
@zpwenjh Thank _you_!

**Comment by ce1adon** [comment]
This PR is ready to review.

**Comment by muralinr** [comment]
@ce1adon @zpwenjh Does this change work for accesses not multiple of 4 in WH? e.g., for C=6, W=H=75, calculated bias and scale for channel 0 depend on the first few elements of channel 1, and the last two entries of channel 5 are never used for bias/scale calculations at all (because 675*75 = 2 mod 4). Can you please check it?

**Comment by ce1adon** [comment]
> @ce1adon @zpwenjh Does this change work for accesses not multiple of 4 in WH? e.g., for C=6, W=H=75, calculated bias and scale for channel 0 depend on the first few elements of channel 1, and the last two entries of channel 5 are never used for bias/scale calculations at all (because 675*75 = 2 mod 4). Can you please check it?

Did you find other failed cases? Would you be able to create an issue and add them in so we can track them?

**Pull request merged**